config/cli_source.go

```diff
@@ -0,0 +1,143 @@
+package config
+
+import (
+	"errors"
+	"strings"
+
+	"github.com/urfave/cli"
+)
+
+// CliSource is an implementation of a Source that reads the key / value pairs
+// from the command line arguments using urfave/cli framework.
+//
+// In urface/cli, there is two ways to define a flag: either define global flags
+// on the `app` or define flags on commands. Flags are then retrieved using the
+// `cli.Context` structure. In the former approach, one can retrieve the flag
+// using `context.GlobalString(key)`. In the latter approach, one can retrieve
+// the flag using `context.String(key)`.
+//
+// Finally, CliSource implements a feature to be able to define any key/value
+// pair on the command line. To use this feature, you need to register a
+// GenericFlag to the `cli.App` or `cli.Command` such as
+//
+//    Flags: []cli.Flag{Name: "generic", Value: &config.GenericFlag{}}
+//
+// This way, one can write any key/pair on the command line using the following
+// format:
+//
+//   ./MyApp --generic "globalKey=value" <command> --generic "key=value"
```

Comment:
I got used to this 'sub-command flags' handling that urfave/cli does. I'm not sure we need the second `--generic`, too.